src/core/services/TransactionService.ts

```diff
@@ -65,7 +65,17 @@ export class TransactionServiceImpl implements TransactionService {
       // const decodedData = contractIface.decodeFunctionData(methodName, unsignedTx.data!.toString());
       // console.log({ decodedData });
 
-      // TODO call contract verification here.
+      const yearn = this.yearnSdk.getInstanceOf(network);
+      if (yearn.services.allowList) {
+        const { success: isValid, error } = await yearn.services.allowList.validateCalldata(
+          contractAddress,
+          unsignedTx.data
+        );
+        if (!isValid) {
+          if (!error) throw new Error('something went wrong but no error was provided');
```

Comment:
Not sure about this error, we could specify more, like "Is not valid with the Allowlist" or whatever, approved tho

Reply:
But we can NOT know if tx is not valid or something exploded on the validation method. Thats why we should not assume that if `isValid` returned false is because tx is not valid. SDK returns isValid = false if there was any kind of error on the method. For example, if the SC call fails due to network acting weird, sdk returns isValid = false, but there was no actual validation. The call to the SC just expoded for some random reason.